packages/hoppscotch-common/src/components/app/Banner.vue

```diff
@@ -0,0 +1,44 @@
+<template>
+  <div
+    class="flex items-center px-4 py-2 text-tiny"
+    :class="bannerColor"
+    :role="bannerRole"
+  >
+    <icon-lucide-info class="mr-2 text-secondaryDark" />
+    <span class="text-secondaryDark">
+      <span v-if="banner.alternateText" class="md:hidden">
+        {{ banner.alternateText }}
+      </span>
+      <span class="<md:hidden">
+        {{ banner.text }}
+      </span>
+    </span>
+  </div>
+</template>
+
+<script setup lang="ts">
+import { computed } from "vue"
+import { BannerContent, BannerType } from "~/services/banner.service"
+
+const props = defineProps<{
+  banner: BannerContent
+}>()
+
+const bannerColor = computed(() => {
+  const bgColors: Record<BannerType, string> = {
+    error: "bg-red-700",
+    warning: "bg-yellow-700",
+    info: "bg-stone-800",
+  }
+  return bgColors[props.banner.type]
+})
```

Comment:
Since `bgColors` is always the same, we should move this out of the `computed` block to prevent this from being recreated for all updates.

```suggestion
const bgColors: Record<BannerType, string> = {
    error: "bg-red-700",
    warning: "bg-yellow-700",
    info: "bg-stone-800",
  }
  
const bannerColor = computed(() => bgColors[props.banner.type])
```

You could even move `bgColors` out into a separate `script` (not setup) block and we can share the same reference across component instances, but since the component is only used once, there is not much difference.

P.S this also applies to `bannerRole` block below.